### PR TITLE
add more test scene for server

### LINKED ISSUE #192: Add unit test to fescar-server module
Add unit test to fescar-server module

### PR DESCRIPTION
**Pull request by leizhiyuan**
### Ⅰ. Describe what this PR did

add test for seata-server

### Ⅱ. Does this pull request fix one issue?

fixes #192

### Ⅲ. Why don't you add test cases (unit test/integration test)? 

add test 

### Ⅳ. Describe how to verify it

test is ok

### CODE REVIEW

File: server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoordinatorTest.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.concurrent.TimeoutException;
+
+public class DefaultCoordinatorTest {
```
Comment (ujjboy) [inline server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoordinatorTest.java]: Add author info.

File: server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoreTest.java [hunk 2]
```diff
@@ -134,12 +135,60 @@ public void commitTest(String xid) throws Exception {
...
-    public void doGlobalCommitTest(String xid) throws Exception {
+    public void doGlobalCommitCommitTest(String xid) throws Exception {
         GlobalSession globalSession = SessionHolder.findGlobalSession(XID.getTransactionId(xid));
+        BranchSession branchSession = new BranchSession();
+        globalSession.addBranch(branchSession);
+        globalSession.changeBranchStatus(branchSession, BranchStatus.PhaseOne_Done);
+
+        core.setResourceManagerInbound(new MockResourceManagerInbound(BranchStatus.PhaseTwo_Committed, BranchStatus.PhaseOne_Done));
+
+
```
Comment (ujjboy) [inline server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoreTest.java]: remove the blank line.

File: server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoreTest.java [hunk 3]
```diff
@@ -134,12 +135,60 @@ public void commitTest(String xid) throws Exception {
...
         core.doGlobalCommit(globalSession, false);
         Assert.assertEquals(globalSession.getStatus(), GlobalStatus.Committed);
     }
 
+
```
Comment (ujjboy) [inline server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoreTest.java]: remove the blank line.

File: server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoordinatorTest.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    }
+
+    @Test(dataProvider = "xidAndBranchIdProviderForCommit")
+    public void branchCommit(String xid, Long branchId) {
+        BranchStatus result = null;
+
+        try {
+            result = defaultCoordinator.branchCommit(BranchType.AT, xid, branchId, resourceId, applicationData);
+        } catch (TransactionException e) {
+            e.printStackTrace();
```
Comment (ujjboy) [inline server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoordinatorTest.java]: Don't use `e.printStackTrace();`

File: server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoreTest.java [hunk 5]
```diff
@@ -134,12 +135,60 @@ public void commitTest(String xid) throws Exception {
...
+    @Test(dataProvider = "xidProvider")
+    public void doGlobalCommitUnretryableTest(String xid) throws Exception {
+        GlobalSession globalSession = SessionHolder.findGlobalSession(XID.getTransactionId(xid));
+        BranchSession branchSession = new BranchSession();
+        globalSession.addBranch(branchSession);
+        globalSession.changeBranchStatus(branchSession, BranchStatus.PhaseOne_Done);
+
+        core.setResourceManagerInbound(new MockResourceManagerInbound(BranchStatus.PhaseTwo_CommitFailed_Unretryable, BranchStatus.PhaseOne_Done));
+
+
```
Comment (ujjboy) [inline server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoreTest.java]: remove the blank line.

File: server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoreTest.java [hunk 6]
```diff
@@ -134,12 +135,60 @@ public void commitTest(String xid) throws Exception {
...
+    @Test(dataProvider = "xidProvider")
+    public void doGlobalCommitExpTest(String xid) throws Exception {
+        GlobalSession globalSession = SessionHolder.findGlobalSession(XID.getTransactionId(xid));
+        BranchSession branchSession = new BranchSession();
+        globalSession.addBranch(branchSession);
+        globalSession.changeBranchStatus(branchSession, BranchStatus.PhaseOne_Done);
+
+        core.setResourceManagerInbound(new MockResourceManagerInbound(BranchStatus.PhaseOne_Timeout, BranchStatus.PhaseOne_Done));
+
+
```
Comment (ujjboy) [inline server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoreTest.java]: remove the blank line.

File: server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoreTest.java [hunk 7]
```diff
@@ -159,12 +208,51 @@ public void rollBackTest(String xid) throws Exception {
...
         GlobalSession globalSession = SessionHolder.findGlobalSession(XID.getTransactionId(xid));
+        BranchSession branchSession = new BranchSession();
+        globalSession.addBranch(branchSession);
+        globalSession.changeBranchStatus(branchSession, BranchStatus.PhaseOne_Done);
+        core.setResourceManagerInbound(new MockResourceManagerInbound(BranchStatus.PhaseTwo_Committed, BranchStatus.PhaseTwo_Rollbacked));
         core.doGlobalRollback(globalSession, false);
         Assert.assertEquals(globalSession.getStatus(), GlobalStatus.Rollbacked);
     }
 
+
```
Comment (ujjboy) [inline server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoreTest.java]: remove the blank line.

Comment (leizhiyuan) [reply server/src/test/java/com/alibaba/fescar/server/coordinator/DefaultCoreTest.java]: ok。done

**Pull request merged**